core/unix/signal_linux_riscv64.c

```diff
@@ -148,10 +149,18 @@ sigill_detected(void *func)
     set_handler_sigact(&act, SIGILL, (handler_t)catch_sigill);
     sigaction_syscall(SIGILL, &act, &old_act);
 
+    /* We use dr_longjmp to exit the SIGILL handle, which skips the signal mask restoring
+     * of OS. Manually save and restore the signal mask here.
```

Comment:
Hmm, looks like there is a mask in dr_jmp_buf_t but only for debug checks: the assumption is that the mask in the frame will be used from within the handler; but there's no convenience routine dr_siglongjmp() (or dr_longjmp_sigmask() to match the dr_setjmp_sigmask() name).  Add a comment like `XXX: Add dr_longjmp_sigmask() to make this easier?`?

Reply:
> Hmm, looks like there is a mask in dr_jmp_buf_t but only for debug checks: the assumption is that the mask in the frame will be used from within the handler; but there's no convenience routine dr_siglongjmp() (or dr_longjmp_sigmask() to match the dr_setjmp_sigmask() name). Add a comment like `XXX: Add dr_longjmp_sigmask() to make this easier?`?

I have a question. In this initialization phase, should the original kernel routine be used for signal handling? DR has not intercepted the signal yet. So I think there is no frame we can use to extract the old mask?

Reply:
There is an init-time signal handler installed for crash signals set in d_r_signal_init().